app/services/proforma_service/convert_task_to_exercise.rb

```diff
@@ -63,5 +64,10 @@ def codeocean_file_from_task_file(file)
       end
       codeocean_file
     end
+
+    def file_type(extension)
+      FileType.find_by(file_extension: extension) ||
+        FileType.create(file_extension: extension, name: extension[1..-1], user: @user, indent_size: 4, editor_mode: 'ace/mode/plain_text')
+    end
```

Comment:
You might be able to use `find_or_create_by` and pass a block which is only evaluated for new items:

```
FileType.find_or_create_by(file_extension: extension) do |filetype|
  filetype.name: extension[1..-1]
  filetype.user: @user, 
  filetype.indent_size: 4
  filetype.editor_mode: 'ace/mode/plain_text'
end
```